cpu/sam0_common/periph/timer.c

```diff
@@ -324,6 +324,22 @@ unsigned int timer_read(tim_t tim)
     return dev(tim)->COUNT.reg;
 }
 
+unsigned int timer_max(tim_t dev)
+{
+    switch (dev) {
+#if TIMER_0_EN
+    case TIMER_0:
+        return TIMER_0_MAX_VALUE;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        return TIMER_1_MAX_VALUE;
+#endif
+    default:
+        return 0;
+    }
```

Comment:
That's not necessary true.
When `timer_set_periodic` is used, arbitrary TOP values can be set at run-time.

Reply:
I'm a bit confused, in what way does `timer_set_periodic` change the maximum compare value?

Reply:
With `TIM_FLAG_RESET_ON_MATCH` the timer will reset when it reaches the set compare value, so any larger value can not be reached anymore.

You *could* use `timer_set_periodic()` with `TIM_FLAG_RESET_ON_MATCH`` and `timer_set_absolute` on the same timer with different channels.

If the the time set with `timer_set_absolute` is > the TOP value set with `timer_set_periodic`, it should be reached. (But reached again when the period timer is disabled)

Reply:
Ahh ok, but you are tight IMO the api refers to the maximum value the counter can hold for that configuration. How would you phrase this for it to be clearer in the api:?

`Get the timers counter size` instead of `Get the timers maximal counter value.`?